### PR TITLE
mgr/dashboard: reorder dashboard side nav bar content

### PR DESCRIPTION
**Pull request by aaSharma14**
[Screencast from 2023-05-18 19-13-28.webm](https://github.com/ceph/ceph/assets/66050535/c8aeb353-e6fd-4ea4-acf2-88ff4404d280)

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/pybind/mgr/dashboard/frontend/src/app/core/navigation/navigation/navigation.component.html [hunk 1]
```diff
@@ -174,6 +157,86 @@
...
+                routerLink="/mgr-modules">Manager Modules</a>
+            </li>
+            <li routerLinkActive="active"
+                class="tc_submenuitem tc_submenuitem_users"
+                *ngIf="permissions.configOpt.read">
+              <a i18n
+                routerLink="/ceph-users">Ceph Users</a>
+            </li>
+          </ul>
+>>>>>>> 92fbd83a70b (mgr/dashboard: reorder dashboard side nav bar content)
```
Comment (nizamial09) [inline src/pybind/mgr/dashboard/frontend/src/app/core/navigation/navigation/navigation.component.html:239]: i guess this is not properly rebased

**Review by nizamial09 — CHANGES_REQUESTED** [review]

**Comment by cloudbehl** [comment]
@aaSharma14 can we have pools inside the cluster nav.
Cluster>
  Hosts
  OSDs
  Pools

File: src/pybind/mgr/dashboard/frontend/src/app/core/navigation/navigation/navigation.component.html (hunk 1 above)
Comment (aaSharma14) [reply src/pybind/mgr/dashboard/frontend/src/app/core/navigation/navigation/navigation.component.html]: Updated the PR

**Comment by aaSharma14** [comment]
> @aaSharma14 can we have pools inside the cluster nav. Cluster> Hosts OSDs Pools

Hey @cloudbehl , updated the PR with this change

**Comment by epuertat** [comment]
Duplicate of https://github.com/ceph/ceph/pull/44153

**Comment by epuertat** [comment]
Hi @aaSharma14 here my high-level comments on this idea, which I fully support btw:
* There's this UX (unwritten) principle about not showing more than ~7 items together (there's this idea on that humans can optimally retain up to ~7 concepts together in their minds). The number itself is not really meaningful, but the fact that we should put an upper bound (something between 5-7) to the amount of items displayed together (not only for remembering when things are, but also for the associated cognitive load). That said:
  * I'd be very cautious about adding another level of nesting (this involves +3 mouse clicks for anything. Instead, I'd consider pulling entries out to another 1st-level item (e.g.: Services, comprising OSDs, Monitors, and Services; or Monitoring).
* One thing it'd be basic on this refactoring would be to just clearly display the 3 types of interfaces to Ceph (or any storage system):
  * Block (we already have that): images, mirroring (could we rename this as multi-cluster or something, to make it more evident what's about?), iSCSI and soon NVMe-oF.
  * Object (I'd put this before Filesystems, since this is the `#2` or even `#1` Ceph use-case) :
  * File System: and I'd include here CephFS and NFS (what matters here is what's the outer use-case to the user, file-like storaage, not whether this uses CephFS internally or RGW).
* Currently this menu only allows to display 1 expanded sub-item at a time, and automatically collapses the other ones. Funny thing is that this is not the practice in Patternfly nor Carbon, but it also requires some extra coding to the trick, so if we disable that, the code would be simpler, and users could retain multiple submenus open at the same time (which could be useful for them if they're regularly accessing pages in different sub-menus, or if they want to keep on screen things like the monitoring alert badges, or the rbd-mirroring warning/error badges).
  * Connected to this, as the menus open/closed are not kept neither in the URL nor the internal page state, whenever you return back to the previous page or navigate in general, the state of the vertical nav-bar is usually not preserved. It'd be interesting to find a way to persist that, as it can be slightly annoying to reopen the same menu items.

**Pull request closed without merging**